components/analytics/compact-dashboard.tsx

```
@@ -0,0 +1,781 @@
+        if (!data) return [];
+        const slices: { name: string; value: number; color: string; key: string; category: string; faded: boolean }[] = [];
+        const colorPool = Object.keys(chartColors).filter(k => k.startsWith('chart')).map(k => chartColors[k as keyof typeof chartColors]);
+
+        // Group unknown_error by message
+        const unknownGroups: Record<string, number> = {};
+        data.errorBots.forEach(bot => {
+            if (bot.status.category === 'unknown_error') {
+                const msg = bot.status?.message ?? 'Unknown error';
+                unknownGroups[msg] = (unknownGroups[msg] || 0) + 1;
+            }
+        });
```

Biome (1.9.4)

[error] Prefer for...of instead of forEach.

forEach may lead to performance issues when working with large arrays. When combined with functions like filter or map, this causes multiple iterations over the same type.

(lint/complexity/noForEach)